WalletWasabi/Blockchain/Keys/KeyManager.cs

```diff
@@ -235,7 +235,25 @@ public WpkhDescriptors GetOutputDescriptors(string password, Network network)
 	private HdPubKeyGenerator? TaprootExternalKeyGenerator { get; set; }
 	private HdPubKeyGenerator? TaprootInternalKeyGenerator { get; }
 
-	public string WalletName => string.IsNullOrWhiteSpace(FilePath) ? "" : Path.GetFileNameWithoutExtension(FilePath);
+	public string WalletName
```

Comment:
This is dangerous. This bypass all the validations on the wallet names allowing to use reserved names (risky as f**k), it allows path injection (it is very limited thanks to the check), it allows to hide wallets file (in linux at least), it ignores the backup wallets, it doesn't do anything to protect consistency and prevent writes during file movement, etc.

Sanitizing the inputs, specially when manipulating files is extremely important. I would be extreme restrictive here.

Reply:
You're so right. I think the new code addresses the problem. Please, check now.

Reply:
It seems it still allow to hide files in linux. Should that be allowed? 
What about the backup mismatch? Should it be renamed too or not?
I think the you have to lock the operation to make sure we don't lose keys in the process.

Reply:
Yes, because you are moving the file. 
Also, why is it that the `KeyManager` has a `WalletName` property? It is `Wallet` what can have a `WalletName`. The `KeyManager` shouldn't know anything about the Wallet's name.